sympy/functions/combinatorial/factorials.py

```diff
@@ -1057,3 +1058,101 @@ def _eval_is_nonnegative(self):
+
+    .. math:: \frac{(k_1+k_2+\cdots+k_m)!}{k_1!\, k_2! \cdots k_m!} = {k_1\choose k_1}{k_1+k_2\choose k_2}\cdots{k_1+k_2+\cdots+k_m\choose k_m}
+
+    The multinomial coefficients have a direct combinatorial interpretation,
+    as the number of ways of depositing n distinct objects into m distinct bins,
+    with k1 objects in the first bin, k2 objects in the second bin, and so on.
+
+    For the sake of convenience for any negative integer `k` this function
+    will return zero.
+
+    To expand the multinomial when `k` contains a symbol, use either
+    ``expand_func()`` or ``expand(func=True)``. The former will keep
+    the polynomial in factored form while the latter will expand the
+    polynomial itself. See examples for details.
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, Rational, multinomial, expand_func, symbols, gamma
+    >>> n = Symbol('n', integer=True, positive=True)
+
+    >>> multinomial(15, 8)
+    490314
+
+    Comparing Coefficents generated with the multinomial function:
+
+    >>> a, b, c = symbols('a b c')
+    >>> ((a+b+c)**3).expand(func=True)
+    a**3 + 3*a**2*b + 3*a**2*c + 3*a*b**2 + 6*a*b*c + 3*a*c**2 + b**3 + 3*b**2*c + 3*b*c**2 + c**3
+
+    The Coefficient when of 2 1 0 corresponding to the second term is 3
+
+    >>> multinomial(2, 1, 0)
+    3
+
+    >>> multinomial(Rational(5, 4), 3)
+    gamma(21/4)/(6*gamma(9/4))
+
+    >>> multinomial(n, 3).rewrite(gamma)
+    gamma(n + 4)/(6*gamma(n + 1))
+
+    >>> multinomial(n, 3).expand(func=True)
+    n**3/6 + n**2 + 11*n/6 + 1
+
+    >>> expand_func(multinomial(n, 3))
+    (n + 1)*(n + 2)*(n + 3)/6
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Multinomial_theorem#Multinomial_coefficients
+
+    .. [2] http://functions.wolfram.com/GammaBetaErf/Multinomial/02/0001/
+
+    """
+
+    @classmethod
+    def eval(self, *args):
+        args = list(map(sympify, args))
+        if len(args) < 2:
+            raise BadArgumentsError("A miniumum of 2 args required")
+        if any(x.is_Integer and x.is_negative for x in args):
+            return S.Zero
+        else:
+            if all(x.is_Integer for x in args):
+                from sympy import binomial
+                if len(args) == 2:
+                    return binomial(sum(args), args[1])
+                return Mul(*[binomial(j, k) for j, k in zip([Add(*args[:i]) for i in range(1, len(args) + 1)], args)])
+            elif all(x.is_number for x in args):
+                from sympy import gamma
+                return gamma(Add(*args) + 1)/Mul(*(gamma(x + 1) for x in args))
+
+    def _eval_rewrite_as_factorial(self, *args, **kwargs):
+        return factorial(Add(*args))/Mul(*(factorial(x) for x in args))
+
+    def _eval_expand_func(self, **hints):
+        from sympy import gamma, expand_func
+        return expand_func(gamma(Add(*self.args) + 1)/Mul(*(gamma(x + 1) for x in self.args)))
+
+    def _eval_rewrite_as_binomial(self, *args, **kwargs):
```

Comment:
I'm not sure this should be here. If it can be rewritten as binomial it will be from the start. And if it wasn't at the outset then it shouldn't be rewritten because assumptions might be violated.

Reply:
I don't get it ? what do you mean ?
I put it there for the same reason as rewrite_as_factorial and rewrite_as_gamma

Reply:
oh wait: if all args are ints and not negative then it can be rewritten, so that check needs to be here, too

Reply:
alright!

Reply:
I believe eval takes care of those checks before ever reaching rewrite ?
```python3
>>> n, k = symbols('n k', integer=False, negative=True)
>>> multinomial(n, k)
multinomial(n, k)
>>> multinomial(n, k).rewrite(binomial)
binomial(k + n, k)
>>> n, k = symbols('n k', integer=True, negative=True)
>>> multinomial(n, k)
0
>>> multinomial(n, k).rewrite(binomial)
0
```

Reply:
`eval` defines the conditions under which args should be written in terms of binomial and gamma.
```python
>>> n, k = symbols('n k', integer=False, negative=True)
>>> multinomial(n, k).rewrite(binomial)
binomial(k + n, k)
```
If that is true then why does `eval` only rewrite as `binomial` if integer is True (and negative is False)?

Reply:
Ohhhh okay that's what you've been saying ... makes perfect sense ... I can eliminate gamma under eval ... just have binomial ... it's more efficient and tht'll solve the rewrite problem too!!!

Reply:
right ?